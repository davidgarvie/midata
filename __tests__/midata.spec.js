const midata = require("..");
const { readFile } = require("fs");
const { resolve } = require("path");

describe("midata", () => {
  let mockData;

  beforeAll(done => {
    const filePath = resolve(__dirname, "../__mocks__/data.csv");
    readFile(filePath, "utf-8", (err, data) => {
      if (!err) {
        mockData = data;
        done();
      }
    });
  });

  test("Should read some midata data", () => {
    expect(midata(mockData)).toMatchSnapshot();
  });
});
