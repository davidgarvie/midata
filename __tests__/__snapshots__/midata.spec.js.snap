// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`midata Should read some midata data 1`] = `
Array [
  Object {
    "Balance": "+£40.00",
    "Date": "05/09/2018",
    "Debit/Credit": "-£10.00",
    "Merchant/Description": "The 3rd Description",
    "Type": "PAYMENTS",
  },
  Object {
    "Balance": "+50.00",
    "Date": "04/09/2018",
    "Debit/Credit": "-50.00",
    "Merchant/Description": "The 2nd Description",
    "Type": "FAST PAYMENT",
  },
  Object {
    "Balance": "+100.00",
    "Date": "03/09/2018",
    "Debit/Credit": "-£20.00",
    "Merchant/Description": "A SECOND MERCHANT",
    "Type": "PAYMENTS",
  },
  Object {
    "Balance": "+£120.00",
    "Date": "02/09/2018",
    "Debit/Credit": "+£25.00",
    "Merchant/Description": "The 1st Description",
    "Type": "CARD PAYMENT",
  },
  Object {
    "Balance": "+95.00",
    "Date": "01/09/2018",
    "Debit/Credit": "-£5.00",
    "Merchant/Description": "A MERCHANT",
    "Type": "DD",
  },
]
`;
